shared-libs/cht-datasource/src/local/target-interval.ts

```diff
@@ -38,4 +39,100 @@ export namespace v1 {
       return doc;
     };
   };
+
+  /** @internal */
+  export const getPage = ({ medicDb: db }: LocalDataContext) => {
+    return async (
+      qualifier: (ReportingPeriodQualifier & ContactUuidsQualifier),
+      cursor: Nullable<string>,
+      limit: number,
+    ): Promise<Page<TargetInterval.v1.TargetInterval>> => {
+      
+      const skip = validateCursor(cursor);
+      const intervalIds = await fetchTargetIntervalIds(limit, skip)(db, qualifier);
+
+      return await fetchTargetIntervalDocs(limit, skip)(db, intervalIds.data);
+    };
+  };
+  
+  const fetchTargetIntervalIds = (limit: number, skip: number) => async (
+    db: PouchDB.Database, 
+    qualifier: (ReportingPeriodQualifier & ContactUuidsQualifier)
+  ) => {
+    const fetchFn = async (
+      limit: number,
+      skip: number
+    ) => {
+      const result = await db
+        .allDocs({ include_docs: false, limit, skip, startkey: `target~${qualifier.reportingPeriod}` });
+      
+      if (result.rows.length === 0) {
+        return [];
+      }
+
+      return result.rows
+        .filter(row => row.doc)
+        .map(row => row.doc as TargetInterval.v1.TargetInterval);
+    };
+
+    const filterFn = (doc: Nullable<Doc>, uuid?: string) => {
+      if (!doc) {
+        if (uuid) {
+          logger.warn(`No target interval found for identifier [${uuid}].`);
+        }
+        return false;
+      }
+
+      for (const contactUuid of qualifier.contactUuids) {
+        if (contactUuid.includes(uuid!)) {
+          return true; 
+        }
+      }
+
+      return false;
+    };
+
+    return fetchAndFilter(
+      fetchFn,
+      filterFn,
+      limit
+    )(limit, skip);
+  };
+
+  const fetchTargetIntervalDocs = (limit: number, skip: number) => async (
+    db: PouchDB.Database, 
+    targetIntervals: TargetInterval.v1.TargetInterval[]
+  ) => {
+    const fetchFn = async (
+      limit: number,
+      skip: number
+    ) => {
+      const result = await db
+        .allDocs({ include_docs: true, limit, skip, keys: targetIntervals.map(t => t._id) });
```

Comment:
```suggestion
    targetIntervalUuids: string[]
  ) => {
    const fetchFn = async (
      limit: number,
      skip: number
    ) => {
      const result = await db
        .allDocs({ include_docs: true, limit, skip, keys: targetIntervalUuids });
```

The last piece we need is to just tweak the parameter here to just be the uuid strings.